projects/observability/src/pages/apis/api-detail/api-detail-breadcrumb.resolver.ts

```diff
@@ -1,101 +1,78 @@
+import { EntityBreadcrumbResolver } from '../../../shared/services/entity-breadcrumb/entity-breadcrumb.resolver';
+import { EntityIconLookupService } from './../../../shared/services/entity/entity-icon-lookup.service';
+import { ApiEntity } from './api-detail.service';
 
 @Injectable({ providedIn: 'root' })
-export class ApiDetailBreadcrumbResolver implements Resolve<Observable<Breadcrumb>> {
-  private readonly specificationBuilder: SpecificationBuilder = new SpecificationBuilder();
+export class ApiDetailBreadcrumbResolver<T extends ApiEntity> extends EntityBreadcrumbResolver<T> {
   protected readonly apiEntityMetadata: EntityMetadata | undefined;
 
   public constructor(
+    timeRangeService: TimeRangeService,
+    graphQlQueryService: GraphQlRequestService,
+    iconLookupService: EntityIconLookupService,
     private readonly navigationService: NavigationService,
-    private readonly timeRangeService: TimeRangeService,
-    private readonly graphQlQueryService: GraphQlRequestService,
     protected readonly breadcrumbService: BreadcrumbsService,
     @Inject(ENTITY_METADATA) private readonly entityMetadataMap: EntityMetadataMap
   ) {
+    super(timeRangeService, graphQlQueryService, iconLookupService);
     this.apiEntityMetadata = this.entityMetadataMap.get(ObservabilityEntityType.Api);
   }
 
   public async resolve(activatedRouteSnapshot: ActivatedRouteSnapshot): Promise<Observable<Breadcrumb>> {
     const id = activatedRouteSnapshot.paramMap.get('id') as string;
-    const parentType = this.resolveParentType();
+    const parentEntityMetadata = this.resolveParentType();
 
     return Promise.resolve(
-      this.fetchEntity(id, parentType).pipe(
-        take(1),
+      this.fetchEntity(id, ObservabilityEntityType.Api).pipe(
+        map(apiEntity => ({
+          ...apiEntity,
+          ...this.getParentPartial(apiEntity, parentEntityMetadata)
+        })),
         switchMap(api => [
-          ...this.getParentBreadcrumbs(api, parentType),
+          ...this.getParentBreadcrumbs(api, parentEntityMetadata),
           this.createBreadcrumbForEntity(api, activatedRouteSnapshot)
         ])
       )
     );
   }
 
-  protected createBreadcrumbForEntity(
-    api: ApiBreadcrumbDetails,
-    activatedRouteSnapshot: ActivatedRouteSnapshot
-  ): Breadcrumb {
+  protected createBreadcrumbForEntity(api: T & Breadcrumb, activatedRouteSnapshot: ActivatedRouteSnapshot): ApiEntity {
     return {
+      ...api,
       label: api.name,
       icon: this.apiEntityMetadata?.icon,
       url: this.breadcrumbService.getPath(activatedRouteSnapshot)
     };
   }
 
-  protected getParentBreadcrumbs(api: ApiBreadcrumbDetails, parentEntityMetadata?: EntityMetadata): Breadcrumb[] {
+  protected getParentBreadcrumbs(api: T & Breadcrumb, parentEntityMetadata?: EntityMetadata): Breadcrumb[] {
     return parentEntityMetadata !== undefined
       ? [
           {
-            label: api.parentName,
+            label: api.parentName as string,
```

Comment:
types are wrong here, shouldn't need to cast

Reply:
Ts is not able to detect the types in the pipe -> map operator

Reply:
What's the type on line 42? Are we sure we're not just dropping it on 59? That map should produce a `T & Partial<Pick<ApiBreadcrumbDetails, 'parentName' | 'parentId'>>` since we're spreading them together (that partial could use a type alias!).

Reply:
(nit btw)

Reply:
Yeah, I am not able to make it work. IMO, `getParentPartial` is making the parentName and parentId optional and then we are required fields in `ApiBreadcrumbDetails`. 

We wouldn't have to cast if we can ensure that following code returns an `ApiBreadcrumbDetails` object
```
 map(apiEntity => ({
          ...apiEntity,
          ...this.getParentPartial(apiEntity, parentEntityMetadata)
        })),
```

Reply:
I can do that by assigning '' instead of undefined but i don't know if it would create any run time issue. So, I am ignoring it for this change.

Reply:
we could just make those two fields optional on `ApiBreadcrumbDetails` I think? They _are_ optional, they're omitted if we can't resolve them, hence the check on line 60 before we access them.